### PR TITLE
Points History and Rewards tab bar styling + Rewards navigation animations

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Overall, made Rewards and Point History look and feel smoother + various other cleanup.
* Made a new `CircleIcon` component to standardize all the repeated circle icons throughout the app
* Fully styled transaction cards in the points history screen and updated copy to match Figma
* Updated transaction card content:
  * Fixed date formatting
  * Renamed the Airtable call to match the new column name 'Points Earned'
  * Added sale total
* Fixed up styling details in the tab view and cleaned up the Rewards heading bar
* Rearranged drawer navigator so the **Rewards page slides up** as a modal to match the Figma prototypes
* Made 'Points History' in the drawer correctly link to Points History (I think the way I did this was a bit jank so I'd really appreciate some more opinions 😅 )

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
The reasoning for adding RootStack with Rewards as a modal came from [this](https://egghead.io/lessons/react-native-setup-a-stacknavigator-as-a-modal-in-react-navigation).

https://stackoverflow.com/questions/44248403/passing-props-with-screen-option-in-drawernavigator

Sample modal interaction [Expo Snack ](https://snack.expo.io/?platform=android&name=createStackNavigator%20%7C%20React%20Navigation&dependencies=%40react-native-community%2Fmasked-view%40%5E0.1.7%2C%40react-navigation%2Fnative%40%5E5.1.3%2C%40react-navigation%2Fbottom-tabs%40%5E5.2.4%2C%40react-navigation%2Fdrawer%40%5E5.3.4%2C%40react-navigation%2Fmaterial-bottom-tabs%40%5E5.1.6%2C%40react-navigation%2Fmaterial-top-tabs%40%5E5.1.6%2C%40react-navigation%2Fstack%40%5E5.2.6%2Creact-native-reanimated%40%5E1.7.0%2Creact-native-safe-area-context%40%5E0.7.3%2Creact-native-screens%40%5E2.4.0&sourceUrl=https%3A%2F%2Freactnavigation.org%2Fexamples%2F5.x%2Fstack-modal-presentation.js)

### Related PRs
Continuing to finalize and clean up details from #42 and #30 

## How to review
* Nitpick my code! I wasn't really sure if it was a good idea to put `CircleIconContainer` in `BaseComponents`, but I didnt' want to have to make a another styled just for that... thoughts?
* Look for styling details I might have missed
* Click around and test the navigation interactions to see if anything breaks with the new RootStack

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
* Make different kinds of points history cards (not just transactions) to match the designs, including reward redeemed, reward unlocked etc.
* Currently, if you switch to the 'My Rewards' tab from opening 'Points History' from the drawer, it will open back to 'My Rewards' if you hit 'Points History' from the drawer again. I decided not to worry about this now based on @anniero98 's recent findings about the updates coming with react-navigation v5.
* Try to figure out how to smooth the transition closing Rewards after it's been opened from the drawer...it would be nice if it slid down but at the moment it just disappears.
* Consider how we may be able to condense the new `CircleIcon` with `NavButton` from #42 
* Cool scrolling animation where 'Healthy Rewards' shrinks and centers upon scrolling :^)
* Look into [Section List](https://reactnative.dev/docs/0.15/sectionlist) for date dividers for points history

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"
* Tested on iPhone 11 Pro Max, and iPhone 8 and it the transactions cards, tab bars and heading seem to be responsive

### Screenshots
[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
* Points History screen
image

* Rewards heading with tab bar
image

* Transaction cards
image

* Full interaction: 
https://www.loom.com/share/945e0d03cab24fcf817938a088cd6069

🥳Rewards finally slides upward!
CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/BaseComponents.js [hunk 1]
```diff
@@ -24,6 +24,17 @@ export const NavButton = styled.TouchableOpacity`
...
+  align-items: center;
+  justify-content: center;
+  height: 40px;
+  width: 40px;
+  padding: 8px;
+  border-radius: 20px;
+  background-color: ${props => props.color || Colors.lighterGreen};
+`;
+
 export const ButtonContainer = styled.TouchableOpacity``;
```
Comment (kennethlien) [inline components/BaseComponents.js]: style thing? i think a lot of our other stuff has the icon vertically centered so maybe we should change this to match that, but @12aschen should chime in

File: components/rewards/RewardsCard.js [hunk 2]
```diff
@@ -1,25 +1,21 @@
...
-            solid
-            size={24}
-            color={Colors.primaryGreen}
-          />
-        </StarIcon>
+        <CircleIcon
+          icon="star"
+          iconColor={Colors.primaryGreen}
+          circleColor={Colors.lightest}
+        />
```
Comment (kennethlien) [inline components/rewards/RewardsCard.js]: ty for the fix here, it was so jank before LOL

File: screens/rewards/RewardsScreen.js [hunk 3]
```diff
@@ -141,18 +144,17 @@ export default class RewardsScreen extends React.Component {
     return (
       <Container>
         <TopTab>
-          <BackButton onPress={() => this.props.navigation.navigate('Stores')}>
+          <BackButton onPress={() => this.props.navigation.goBack()}>
```
[the remark this answers is not in the document]
Comment (wangannie) [reply screens/rewards/RewardsScreen.js]: Yep! I changed it to goBack because it would reload the entire map when I closed it from the drawer and that got really slow.

File: components/rewards/RewardsHome.js [hunk 4]
```diff
@@ -23,35 +23,38 @@ function createList(N) {
...
-          ))}
-        </AvailiableRewardsContainer>
-      </ScrollView>
-    </View>
+    <ScrollView style={{ marginLeft: 16, paddingRight: 16 }}>
+      <RewardsProgressContainer>
+        <Overline style={{ marginTop: 24, marginBottom: 12 }}>
+          Reward Progress
+        </Overline>
+        <Title style={{ marginBottom: 2 }}>
```
[the remark this answers is not in the document]
Comment (wangannie) [reply components/rewards/RewardsHome.js]: I think Kenneth and I agreed on abandoning that when the alignment of 2 different line heights was getting really annoying 😶

File: components/BaseComponents.js (hunk 1 above)
Comment (wangannie) [reply components/BaseComponents.js]: Not sure what you mean—vertically centered to what? Lines 29 & 30 should make it centered.
